junit-platform-commons/src/main/java/org/junit/platform/commons/util/CloseablePath.java

```diff
@@ -27,10 +28,19 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Function;
 
+import org.apiguardian.api.API;
+
 /**
+ * <h2>DISCLAIMER</h2>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
  * @since 1.0
  */
-final class CloseablePath implements Closeable {
+@API(status = INTERNAL, since = "1.12")
+public final class CloseablePath implements Closeable {
```

Comment:
Should be moved to the `org.junit.platform.commons.support.scanning` package because it's only used there and kept package-private.